Sources/MongoSwift/MongoError.swift

```diff
@@ -97,20 +97,42 @@ public struct WriteError: Codable {
     /// An integer value identifying the error.
     public let code: ServerErrorCode
 
+    /// A human-readable string identifying the error.
+    public let codeName: String
+
     /// A description of the error.
     public let message: String
 
     private enum CodingKeys: String, CodingKey {
         case code
+        case codeName
         case message = "errmsg"
     }
+
+    // TODO: can remove this once SERVER-36755 is resolved
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.code = try container.decode(ServerErrorCode.self, forKey: .code)
+        self.message = try container.decode(String.self, forKey: .message)
+        self.codeName = try container.decodeIfPresent(String.self, forKey: .codeName) ?? ""
+    }
+
+    // TODO: can remove this once SERVER-36755 is resolved
+    public init(code: ServerErrorCode, codeName: String, message: String) {
```

Comment:
oops I approved but realized I still had a couple questions.
why does the server bug necessitate having this initializer?
and why does the initializer need to be `public`?

Reply:
It needs to exist because explicitly adding `init(from: Decoder)` prevents the synthesized initializer from being synthesized. Good catch though, it doesn't need to be public, fixed.